slashing-protection/referencetests/src/test/java/tech/pegasys/web3signer/slashingprotection/ReferenceTestRunner.java

```diff
@@ -0,0 +1,155 @@
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.web3signer.slashingprotection;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import tech.pegasys.web3signer.slashingprotection.dao.ValidatorsDao;
+import tech.pegasys.web3signer.slashingprotection.interchange.InterchangeModule;
+import tech.pegasys.web3signer.slashingprotection.model.TestFileModel;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import com.fasterxml.jackson.databind.MapperFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.io.Resources;
+import com.opentable.db.postgres.embedded.EmbeddedPostgres;
+import dsl.SignedArtifacts;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.tuweni.bytes.Bytes;
+import org.flywaydb.core.Flyway;
+import org.jdbi.v3.core.Jdbi;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+
+public class ReferenceTestRunner {
+
+  private static final Logger LOG = LogManager.getLogger();
+
+  private static final ObjectMapper objectMapper =
+      new ObjectMapper()
+          .registerModule(new InterchangeModule())
+          .enable(MapperFeature.ACCEPT_CASE_INSENSITIVE_ENUMS);
+  private final ValidatorsDao validators = new ValidatorsDao();
+  private EmbeddedPostgres slashingDatabase;
+
+  @Before
+  void setup() throws IOException {
+    slashingDatabase = EmbeddedPostgres.start();
+    final Flyway flyway =
+        Flyway.configure()
+            .locations("/migrations/postgresql/")
+            .dataSource(slashingDatabase.getPostgresDatabase())
+            .load();
+    flyway.migrate();
+  }
+
+  @After
+  void cleanup() {
+    try {
+      slashingDatabase.close();
+    } catch (final IOException e) {
+      LOG.error("Failed to close database", e);
+    }
+  }
+
+  @Test
+  @Disabled("Until interchange import is available")
+  void execute() throws IOException {
+    final URL refTestPath =
+        Resources.getResource(
+            Path.of("slashing-protection-interchange-tests", "tests", "generated").toString());
+
+    final Path testFilesPath = Path.of(refTestPath.getPath());
+    try (Stream<Path> testFiles = Files.list(testFilesPath)) {
+      testFiles.forEach(this::executeFile);
+    }
+  }
+
+  private void executeFile(final Path inputFile) {
+    LOG.info(inputFile.toString());
+    try {
+      final TestFileModel model = objectMapper.readValue(inputFile.toFile(), TestFileModel.class);
+      final String interchangeContent =
+          objectMapper.writeValueAsString(model.getInterchangeContent());
+
+      setup();
```

Comment:
Why does setup need to be called? Isn't it being called by the junit framework?

Reply:
removed